w3-blob.md

```diff
@@ -563,6 +565,224 @@ type LocationCommitment = {
 }
 ```
 
+## List Blob
+
+Authorized agent MAY invoke `/space/content/list/blob` capability on the [space] subject (`sub` field) to list Blobs added to it at the time of invocation.
+
+### List Blob Invocation Example
+
+Shown Invocation example illustrates Alice requesting a page of the list of blobs stored on their space.
+
+```js
+{
+  "cmd": "/space/content/list/blob",
+  "sub": "did:key:zAlice",
+  "iss": "did:key:zAlice",
+  "aud": "did:web:web3.storage",
+  "args": {
+    // cursor where to start listing from
+    "cursor": 'cursor-value-from-previous-invocation',
+    // size of page
+    "size": 40,
+  }
+}
+```
+
+### List Blob Receipt Example
+
+Shows an example receipt for the above `/space/content/list/blob` capability invocation.
+
+> ℹ️ We use `// "/": "bafy..` comments to denote CID of the parent object.
+
+```js
+{ // "/": "bafy..list",
+  "iss": "did:web:web3.storage",
+  "aud": "did:key:zAlice",
+  "cmd": "/ucan/assert/result"
+  "sub": "did:web:web3.storage",
+  "args": {
+    // refers to the invocation from the example
+    "ran": { "/": "bafy..list" },
+    "out": {
+      "ok": {
+        // cursor where to start listing from on next call
+        "cursor": "cursor-value-for-next-invocation",
+        // size of the list
+        "size": 40,
+        "results": [
+          {
+            "insertedAt": "2024-04-16T15:49:22.638Z",
+            "blob": {
+              "size": 100,
+              "content": { "/": { "bytes": "mEi...sfKg" } },
+            }
+          },
+          // ...
+        ]
+      }
+    },
+    // set of tasks to be scheduled.
+    "next": []
+  }
+}
+```
+
+### List Blob Capability
+
+#### List Blob Capability Schema
+
+```ts
+type ListBlob = {
+  cmd: "/space/content/list/blob"
+  sub: SpaceDID
+  args: {
+    cursor?: string
+    size?: number
+  }
+}
+```
+
+##### List Cursor
+
+The optional `args.cursor` MAY be specified in order to paginate over the list of the added Blobs.
+
+##### List Size
+
+The optional `args.size` MAY be specified to signal desired page size, that is number of items in the result.
+
+### List Blob Receipt
+
+#### List Blob Receipt Schema
+
+```ts
+type ListBlobReceipt = {
+  out: Result<ListBlobOk, ListBlobError>
+  next: []
+}
+
+type ListBlobOk = {
+  cursor?: string
+  before?: string
+  after?: string
+  size: number
+  results: ListBlobItem
+}
+
+type ListBlobItem = {
```

Comment:
Don't want to complicate things here so it's not a request, more of suggestion with an expectation that we'll probably do it sometime in the future instead.

I think we should stop having lists like and instead just return list of receipts instead, we can let client deal with getting / caching receipts locally and deriving timestamps from when receipt was issued.

Reply:
Yes, actually @alanshaw and I also talked about shifting into that direction later on, and use more the pattern you described recently on a RFC ("our databases should be considered more an index over our receipts"). But for MVP let's make it like store and iterate on this when possible